### DISCUSSION TITLE
How to detect being disconnected by a mod?

### DISCUSSION THREAD

**Discussion by SuperCuber** [serenity-rs/songbird#71] [Q&A] [answered]
Right now I have some cleanup code in my disconnect command (that clears the song queue for example), but I also want to execute it when a mod presses rightclick and disconnects the bot from the channel. How do I hook up an event handler to this? I found [CoreEvent](https://docs.rs/songbird/0.1.6/songbird/events/enum.CoreEvent.html) but it doesn't seem that there even exists a disconnect event - unless the `ClientDisconnect` event applies to the bot itself and not only to other participants.

**Comment by FelixMcFelix** [comment, marked as answer]
Currently we don't have an event type for this, but it should be a doable (non-breaking) addition. I think in the meantime, your best option is to listen into (assuming you're on serenity) [`VoiceStateUpdate`](https://docs.rs/serenity/0.10.5/serenity/model/event/struct.VoiceStateUpdateEvent.html)s—an update with your bot's `UserId` and `None` for its `channel_id` is the relevant trigger.

**Reply by SuperCuber** [reply]
Thanks, looks like this works! Seems like I didn't need to use that struct directly though - I overrode a function in the EventHandler trait.
